### ISSUE TITLE
[Feature Request] Export Logits from network

### ISSUE THREAD

**Issue by Mazecreator** [reinforceio/tensorforce#293]
For debugging of reward space and network structure it would be good to export the raw logits for at a minimum the Classification distribution.  Getting a simple action back does not really show the degree of learning for the network, adding the logits show some additional details such as separation of the classes and noise immunity.  For my personal approach, I don't see any benefit for exporting the logits within training, but usually a loop over the Validation & Test data sets.  I have made some modifications to export the logits for testing.  Here is a simple loop I run within the episode_finished callback:

        state=testenv.reset(True)
        IsProcessingData = True

        additional = list()        

        while IsProcessingData:
            action, logits=r.agent.act(state,deterministic=True, include_logits=True)
            state, terminal, reward = testenv.execute(action)
            new_data = logits[0].tolist()+[action]+[reward]+[terminal]+state["state1"].tolist()+state["start2"].tolist()
            additional.append(np.array(new_data))            
            if terminal:
                if testenv.env.is_over(1):
                    IsProcessingData = False
                else:
                    state=testenv.reset()
    
The goal here was to make the .act() method backward compatible with the rest of the library and only move data around which could impact training/inference performance when the include_logits flag is set.  I will create a PR to show what I have done and start a discussion on approach.

**Comment by Mazecreator** [comment]
@michaelschaarschmidt , I am a bit confused due to the "probability" layer at the end if I want the Logits or the embeddings.  I almost wonder if I should return a tuple of both values?
eg: (logits, embeddings)

**Comment by Mazecreator** [comment]
PR is up for review:  Let me know if there are other classes that need to be modified.  I could only find the distribution_model to hold the logits but sometimes they are masked.
https://github.com/reinforceio/tensorforce/pull/294

**Comment by AlexKuhnle** [comment]
PR is merged.
